lib/performance_data/test_helpers/statistics.rb

```diff
@@ -0,0 +1,72 @@
+require 'gds_api/test_helpers/performance_platform/data_out'
+
+module PerformanceData
+  module TestHelpers
+    module Statistics
+      include GdsApi::TestHelpers::PerformancePlatform::DataOut
+
+      PP_DATA_OUT_ENDPOINT = "http://www.performance.service.gov.uk".freeze
+
+      def params(slug, collect, group_by, is_multipart)
+        params = {
+            collect: collect,
+            group_by: group_by,
+            end_at: Date.today.to_time.getutc.iso8601,
+            duration: 42,
+            period: 'day'
+        }
+
+        filter_param = is_multipart ? :filter_by_prefix : :filter_by
+        params[filter_param] = 'pagePath:' + slug
+
+        params.to_query
+      end
+
+      def search_terms_params(slug)
+        params(slug, 'searchUniques:sum', 'searchKeyword', false)
+      end
+
+      def searches_params(slug, is_multipart = false)
+        params(slug, 'searchUniques:sum', 'pagePath', is_multipart)
+      end
+
+      def page_views_params(slug, is_multipart = false)
+        params(slug, 'uniquePageviews:sum', 'pagePath', is_multipart)
+      end
+
+      def problem_reports_params(slug, is_multipart = false)
+        params(slug, 'total:sum', 'pagePath', is_multipart)
+      end
+
+      def stub_performance_platform_has_slug(slug, response)
+        stub_search_terms(slug, response[:search_terms])
+        stub_searches(slug, false, response[:searches])
+        stub_page_views(slug, false, response[:page_views])
+        stub_problem_reports(slug, false, response[:problem_reports])
+      end
+
+      def stub_performance_platform_has_slug_multipart(slug, response)
+        is_multipart = true
+
+        stub_search_terms(slug, response[:search_terms])
+        stub_searches(slug, is_multipart, response[:searches])
+        stub_page_views(slug, is_multipart, response[:page_views])
+        stub_problem_reports(slug, is_multipart, response[:problem_reports])
+      end
+
+      def stub_performance_platform_has_no_data_for_slug(slug)
+        stub_request(:get, "#{PP_DATA_OUT_ENDPOINT}/data/govuk-info/searchTerms?#{search_terms_params(slug)}").
```

Comment:
If we decide not to though, I think we could drop the `xxx_params` methods and simplify the stubs to: 

```ruby
stub_request(:get, "#{PPR_DATA_OUT_ENDPOINT}/data/govuk-info/searchTerms").with(query: hash_including(filter_by: slug)).to_return(status: 404, headers: { content_type: "application/json" })
```

It's a shorter definition so there's less for us to read to understand, and we don't really care what all the other params are; just that any request including this slug should be a 404.